tiledb/sm/query/query_condition.cc

```diff
@@ -1577,25 +1630,56 @@ void QueryCondition::apply_ast_node_sparse(
     const uint64_t buffer_offsets_el =
         tile_offsets.size() / constants::cell_var_offset_size;
 
-    // Iterate through each cell.
-    for (uint64_t c = 0; c < buffer_offsets_el; ++c) {
-      // Check the previous cell here, which breaks vectorization but as this
-      // is string data requiring a strcmp which cannot be vectorized, this is
-      // ok.
-      const uint64_t buffer_offset = buffer_offsets[c];
-      const uint64_t next_cell_offset =
-          (c + 1 < buffer_offsets_el) ? buffer_offsets[c + 1] : buffer_size;
-      const uint64_t cell_size = next_cell_offset - buffer_offset;
-
-      // Get the cell value.
-      const void* const cell_value = buffer + buffer_offset;
-
-      // Compare the cell value against the value in the value node.
-      const bool cmp = BinaryCmp<T, Op>::cmp(
-          cell_value, cell_size, condition_value_content, condition_value_size);
-
-      // Set the value.
-      result_bitmap[c] = combination_op(result_bitmap[c], cmp);
+    // We need to repeat code/have two for loops in order to support
+    // vectorization. Iterate through each cell.
+    if constexpr (
+        std::is_same_v<CombinationOp, QCMax<BitmapType>> && nullable::value) {
```

Comment:
I think you might be able to move this conditional down to where the value is set and prevent duplicated code... Since it is a constexpr, it will get optimized out depending on the template arguments and will not break vectorization if the CombinationOp is the one we use for AND.